### DISCUSSION TITLE
fan out config via push, not pull

### DISCUSSION THREAD

**Discussion by mdtancsa** [zrepl/zrepl#672] [Q&A] [answered]
Forgive the simple question, but I am having a hard time coming up with the right server config for a scenario where I have one server replicating to multiple destinations via push jobs. I see the pull example in the docs, but not a push.  I have server 
A, X,Y where X and Y where I want to backup TO. On Server A, I have the following config. Am I missing something important here ?

[fan.yaml.txt](https://github.com/zrepl/zrepl/files/10428574/fan.yaml.txt)

Thanks!

**Comment by kpfleming** [comment]
If possible, please repost the config and use the 'Add code' button in the toolbar, so that the formatting won't be lost.

Your config looks fine. If you're not getting the results you want, please describe the results you are getting.

**Reply by mdtancsa** [reply]
Thanks, I updated the post with the config in a .txt file as the "Add code" killed the indents.  I just wanted to make sure if I need to create / specify any differences in the configs per destination server to make them "unique" as implied in the documentation

**Comment by mdtancsa** [comment]
How do you schedule the push ?  If I do a manual signal wakeup, it sends, but it never sends after that ? If I add an interval option, I get 
% zrepl configcheck
could not parse config: yaml: unmarshal errors:
  line 61: field interval not found in type config.PushJob

**Reply by kpfleming** [reply]
That is correct, `push` jobs do not have schedules of their own. You need to use some other tool to trigger them via signal wakeups.

**Comment by mdtancsa** [comment, marked as answer]
OK, getting farther. I put in some hooks on the snapshot process that fires to wake up the job.  But I guess there is no easy way through the existing environment variables to figure out "OK, I have done the last snapshot for this interval, I can now send my wakeup signal."

In my rudimentary hook script, I cant just fire from the post_snapshot as that gets fired per dataset.  I need to send it once the job is complete. Or better yet, is there a way from the CLI to test whether or not a job is already woken up ? Otherwise it creates a lot of spew in the logs as the hook 'fails'

**Reply by kpfleming** [reply]
At the moment the answer is no; the hook script is fired for each snapshot that is created, not for the completion of the 'snap' job.

**Reply by problame** [reply]
Yeah, I’ve seen multiple requests for this kind of use case recently and plan to address this in one of the next zrepl releases.

**Reply by mdtancsa** [reply]
Thanks guys, I have it working the way I want for now with a crontab entry.

**Reply by zakariaboualaid** [reply]
@problame I am curious if this is now addressed?
